### PR TITLE
internal/utils: fix parsing of user versions which include image path suffix

### LINKED ISSUE #692: Cli 0.10.1 breaks the parsing for specific images.
## Bug report

**General Information**

When enabling Hubble with the specific Service Mesh beta images, the parsing of the image name is failing and thus causing the `hubble-relay` pod being not deployed.

This is a regression from `0.10.0` where it works.

- Cilium CLI version (run `cilium version`): 0.10.1

**How to reproduce the issue**

- Install lastest 0.10.1
- Follow the [service mesh beta installation instructions](https://github.com/cilium/cilium-service-mesh-beta/blob/main/INSTALLATION.md)
- Run `cilium hubble enable`

This will hang and produce the output below:
```
⌛ Waiting for Cilium to become ready before deploying other Hubble component(s)...
🔑 Generating certificates for Relay...
✨ Deploying Relay from quay.io/cilium/hubble-relay:-service-mesh:v1.11.0-beta.1...
```
```
  Warning  InspectFailed  4s (x3 over 16s)  kubelet            Failed to apply default image tag "quay.io/cilium/hubble-relay:-service-mesh:v1.11.0-beta.1": couldn't parse image reference "quay.io/cilium/hubble-relay:-service-mesh:v1.11.0-beta.1": invalid reference format
  Warning  Failed         4s (x3 over 16s)  kubelet            Error: InvalidImageName
```

**Workaround**
Running the command with the correct image path: `cilium hubble enable --relay-version -service-mesh:v1.11.0-beta.1`

### PR DESCRIPTION
**Pull request by twpayne**
Fixes #692.

Signed-off-by: Tom Payne <tom@isovalent.com>

### CODE REVIEW

**Comment by twpayne** [comment]
Oops, the code already parsed such versions correctly. The bug is elsewhere. Reverting to draft.

**Comment by twpayne** [comment]
OK, should be actually fixed now 🙈 Ready for review.

File: hubble/relay.go [hunk 1]
```diff
@@ -237,7 +238,14 @@ disable-server-tls: true
...
 func (k *K8sHubble) relayImage(imagePathMode utils.ImagePathMode) string {
-	defaultImage := defaults.RelayImage + ":" + k.ciliumVersion
+	var defaultImage string
+	if strings.Contains(k.ciliumVersion, ":") {
+		// The Cilium version already contains a ":" (separating a suffix and a
+		// version), so append it without the ":".
+		defaultImage = defaults.RelayImage + k.ciliumVersion
+	} else {
+		defaultImage = defaults.RelayImage + ":" + k.ciliumVersion
+	}
```
[the remark this answers is not in the document]
Comment (twpayne) [reply hubble/relay.go]: Ah, I see. PR updated to match your suggestion. PTAL.

**Review by tklauser — APPROVED** [review]
LGTM, thanks Tom!

I've also tested `cilium` built from this PR with the instructions in https://github.com/cilium/cilium-service-mesh-beta/blob/main/INSTALLATION.md and it worked as expected.

**Pull request merged**